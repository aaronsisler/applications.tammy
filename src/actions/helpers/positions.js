export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
