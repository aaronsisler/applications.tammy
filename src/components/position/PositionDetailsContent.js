import React from 'react';
import PropTypes from 'prop-types';

export default class PositionDetailsContent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { position } = this.props;
        return (
            <div className="position_details_content">
                <div className="position_details_content__description">
                    <span>Description:</span>
                    <div>{position.content.description}</div>
                </div>
                {position.content.responsibilities &&
                    <div className="position_details_content__responsibilities">
                        <span>Responsibilities:</span>
                        <ul className="position_details_content__list">
                            {position.content.responsibilities.map((responsibility, index) =>
                                <li key={index}>{responsibility}</li>
                            )}
                        </ul>
                    </div>
                }
                {position.content.requirements &&
                    <div className="position_details_content__requirements">
                        <span>Requirements:</span>
                        <ul className="position_details_content__list">
                            {position.content.requirements.map((requirement, index) =>
                                <li key={index}>{requirement}</li>
                            )}
                        </ul>
                    </div>
                }
            </div>
        );
    }
}

PositionDetailsContent.propTypes = {
    position: PropTypes.object.isRequired,
};

