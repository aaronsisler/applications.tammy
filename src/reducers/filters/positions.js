const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
