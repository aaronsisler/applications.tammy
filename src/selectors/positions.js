// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
