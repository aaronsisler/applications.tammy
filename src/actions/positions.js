import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
