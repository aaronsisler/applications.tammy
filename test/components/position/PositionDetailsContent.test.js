import React from 'react';
import { shallow } from 'enzyme';
import PositionDetailsContent from 'Position/PositionDetailsContent';
import positions from '../../fixtures/positions';

describe('PositionList', () => {
    it('should render PositionDetailsContent correctly', () => {
        const [position] = positions;
        const wrapper = shallow(
            <PositionDetailsContent
                position={position}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
