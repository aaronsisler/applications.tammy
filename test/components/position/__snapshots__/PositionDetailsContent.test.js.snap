// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionList should render PositionDetailsContent correctly 1`] = `
<div
  className="position_details_content"
>
  <div
    className="position_details_content_wrapper"
  >
    <div
      className="position_details_content_wrapper__header"
    >
      Description:
    </div>
    <div>
      first description
    </div>
  </div>
  <div
    className="position_details_content_wrapper"
  >
    <div
      className="position_details_content_wrapper__header"
    >
      Responsibilities:
    </div>
    <ul
      className="position_details_content__list"
    >
      <li
        key="0"
      >
        Responsibility 1
      </li>
      <li
        key="1"
      >
        Responsibility 2
      </li>
    </ul>
  </div>
  <div
    className="position_details_content_wrapper"
  >
    <div
      className="position_details_content_wrapper__header"
    >
      Requirements:
    </div>
    <ul
      className="position_details_content__list"
    >
      <li
        key="0"
      >
        Skill 1
      </li>
      <li
        key="1"
      >
        Skill 2
      </li>
    </ul>
  </div>
</div>
`;
